src/Umbraco.Web.UI.Client/src/common/directives/components/forms/checkintegeronly.directive.js

```diff
@@ -0,0 +1,28 @@
+﻿
+/**
+* @ngdoc directive
+* @name umbraco.directives.directive:checkIntegerOnly
+* @restrict A
+* @description Used in conjunction with type='number' input fields to check if a field should accept integers only. 
+*              This condition is optional based on the property editor configuration specifying that the number is expected to be an integer.
+**/
+function checkIntegerOnly($parse) {
+    return {
+        restrict: "A",
+        require: "ngModel",
+
+        link: function (scope, elem, attrs, ctrl) {
+            if (scope.$parent.model.config && scope.$parent.model.config.integer) {
+                elem.on('keydown', function (event) {
+                    if ([110, 190].indexOf(event.which) > -1) { // Check for decimal point
```

Comment:
What does this mean please? Maybe add a code comment, I suspect this means it's checking for a comma or a period? Maybe this can be changed to a regex that only allows numbers, decimal points in the rest of the world can be different ([though rare](https://en.wikipedia.org/wiki/Decimal_separator#Unicode_characters)).